.github/workflows/docker-build.yml

```diff
@@ -1,38 +1,56 @@
-name: Docker CI workflow 
+﻿name: Docker CI workflow
 
 on:
   push:
     branches: ["main"]
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
   pull_request:
     branches: ["main"]
+    types:
+      - opened
+      - reopened
+      - synchronize
 
 jobs:
-
   docker-build:
     runs-on: ubuntu-latest
-    name: Run docker build  
+    name: Run docker build
     steps:
-      
       - uses: actions/checkout@v4
 
       - name: Docker Login
         uses: docker/login-action@v3.0.0
         with:
           username: ${{ secrets.DOCKER_USER }}
           password: ${{ secrets.DOCKER_TOKEN }}
+      
+      - name: Docker Login
+        uses: docker/login-action@v3.0.0
+        with:
+          registry: ghcr.io
+          username: ${{ secrets.GH_USER }}
+          password: ${{ secrets.GH_TOKEN }}
 
       - name: Define docker image name
         id: define_docker_image
         run: |
-          DOCKER_TAG="testcandidates-$(date +%Y%m%d%H%M%S)"
-          echo DOCKER_IMAGE="planthordev/identityserver:$DOCKER_TAG" >> "$GITHUB_OUTPUT"
+          DOCKER_TAG=""
+          if [[ $GITHUB_REF == 'refs/heads/main' ]]; then
+            DOCKER_TAG="latest"
+            echo DOCKER_IMAGE="ghcr.io/planthor/planthoridentityserver:$DOCKER_TAG" >> "$GITHUB_OUTPUT"
+          elif [[ $GITHUB_REF =~ ^refs/tags/* ]]; then
+            DOCKER_TAG="${{  github.ref_name }}"
+            echo DOCKER_IMAGE="ghcr.io/planthor/planthoridentityserver:$DOCKER_TAG" >> "$GITHUB_OUTPUT"
+          else 
+            DOCKER_TAG="${{ github.head_ref }}"
+            echo DOCKER_IMAGE="planthordev/identityserver:$DOCKER_TAG" >> "$GITHUB_OUTPUT"
+          fi
 
       - name: Build and push Docker images
```

Comment:
Does this mean we will create new image and push every time we create a PR, even though it is not merged? 
If that is the case, then this Github Action should be run on Push only.

Reply:
In this case, I would like to create an image for each branch to have a pull request. The developer needs to test their feature as container. I think its okay then we could have a clean house keeping job in the future

Reply:
Then we should create image and push to registry for "on push" only, we don't have to create image and push with "on pull_request" right ?

Reply:
Okay will remove feature branche case